File: python/14_arrival_of_the_general.py
n = int(input().strip())
arr = list(map(int, input().split()))
imax = arr.index(max(arr))
imin = n - 1 - arr[::-1].index(min(arr))
moves = imax + (n - 1 - imin)
if imax > imin:
    moves -= 1
print(moves)

File: cpp/14_arrival_of_the_general.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){ int n; cin>>n; int h, imax=0, imin=0; int maxh=-1, minh=1e9; for(int i=1;i<=n;i++){ cin>>h; if(h>maxh){ maxh=h; imax=i; } if(h<=minh){ minh=h; imin=i; } }
    int moves = (imax-1) + (n - imin);
    if(imax>imin) moves--;
    cout<<moves<<"\n";
}
